pypiserver/_app.py

```diff
@@ -53,15 +58,8 @@ def protector(*args, **kwargs):
 @app.hook("before_request")
 def log_request():
     log.info(config.log_req_frmt, request.environ)
-
-
-@app.hook("before_request")
-def print_request():
-    parsed = urlparse(request.urlparts.scheme + "://" + request.urlparts.netloc)
-    request.custom_host = parsed.netloc
-    request.custom_fullpath = (
```

Comment:
I see the new tests might be failing on [`custom_fullpath` access in `core.py:L17`](https://github.com/pypiserver/pypiserver/blob/cf424c982d8c66b8ee38984b4c61b8340ddece28/pypiserver/core.py#L15-L17)

Reply:
@oz123 by moving `` to `_app.py` (and adjusting tests accordingly) like this: 

```
# Already added to `_app.py`
def request_fullpath(request):
    parsed = urlparse(request.urlparts.scheme + "://" + request.urlparts.netloc)
    return parsed.path.rstrip("/") + "/" + request.fullpath.lstrip("/")

# Moved from `core.py`
def get_bad_url_redirect_path(request, project):
    """Get the path for a bad root url."""
    uri = request_fullpath(request)
    if uri.endswith("/"):
        uri = uri[:-1]
    uri = uri.rsplit("/", 1)[0]
    project = quote(project)
    uri += f"/simple/{project}/"
    return uri
```

I think that should fix it 🤔